### PR TITLE
Add typehint for Factory

### PR DESCRIPTION
**Pull request by barryvdh**
So that an IDE can autocomplete the calls.

### CODE REVIEW

**Comment by JellyBool** [comment]
Nice job ! @barryvdh

**Comment by spyric** [comment]
It was before #3876 and closed by Taylor

**Comment by GrahamCampbell** [comment]
This is not a typehint, this is just a workaround for an IDE. We shouldn't need to change code at all for an IDE. That defeats.

**Comment by GrahamCampbell** [comment]
Meh, leaving open incase Taylor wants to reconsider.

:-1: from me though.

**Comment by barryvdh** [comment]
We're not changing code, it's just basically impossible for an IDE to detect where the $factories is coming from.
Even for us humans, we can't tell that by just looking at the code..

**Comment by taylorotwell** [comment]
I'll merge this one but I really don't want to get carried away with these IDE work-arounds.

File: database/factories/ModelFactory.php [hunk 1]
```diff
@@ -10,7 +10,7 @@
 | database. Just tell the factory how a default model should look.
 |
 */
-
+/** @var $factory \Illuminate\Database\Eloquent\Factory */
```
Comment (arcanedev-maroc) [inline database/factories/ModelFactory.php]: This typehint must be inverted, like this:

```
/** @var \Illuminate\Database\Eloquent\Factory $factory */
```

Comment (barryvdh) [inline database/factories/ModelFactory.php]: Ah yeah,  https://phpdoc.org/docs/latest/references/phpdoc/tags/var.html
Whoops

**Comment by arcanedev-maroc** [comment]
@taylorotwell This *IDE work-arounds* may saves you a lot of time (autocomplete, navigate-to-method, documentation, etc).

**Comment by taylorotwell** [comment]
So do we need to fix this? The type-hint is wrong format?

**Comment by arcanedev-maroc** [comment]
@taylorotwell PR #3988

**Pull request merged**